dht/token.hh

```diff
@@ -98,10 +98,15 @@ public:
     }
 
     bytes data() const {
-        auto t = net::hton(_data);
-        bytes b(bytes::initialized_later(), sizeof(_data));
-        std::copy_n(reinterpret_cast<int8_t*>(&t), sizeof(_data), b.begin());
-        return b;
+        return write_bytes_be(_data);
+    }
+
+    bytes unsigned_data() const {
+        // Map range from [-2^63, 2^63-1] to [0, 2^64-1]. Written this way 
+        // (subtracting int64_t min) to avoid C++ undefined behaviour.
+        uint64_t unsigned_value = static_cast<uint64_t>(_data);
+        unsigned_value -= static_cast<uint64_t>(std::numeric_limits<int64_t>::min());
```

Comment:
Is this transformation the same as "bias()" in dht/token.hh? Would have been nice if we had this function once in a header file, with decent comments. But we don't have to do it now.